invenio_rdm_records/resources/serializers/marcxml/schema.py

```diff
@@ -470,6 +451,24 @@ def get_additional_descriptions(self, obj):
 
         return additional_descriptions or missing
 
+    def get_languages(self, obj):
+        """Get languages."""
+        languages = obj["metadata"].get("languages")
+
+        if languages:
+            return [{"a": language["id"]} for language in languages]
+
+        return missing
+
+    def get_references(self, obj):
+        """Get references."""
+        references_list = obj["metadata"].get("references", [])
+
+        if references_list:
+            return [{"x": reference["identifier"]} for reference in references_list]
```

Comment:
This should be the reference string instead of the identifier (or if possible both identifier and reference string) - depending on what we had in legacy.